### DISCUSSION TITLE
How can I fetch data for a page meta?

### DISCUSSION THREAD

**Discussion by cawa-93** [nuxt/nuxt#17549] [Questions] [answered]
I have a project where many pages use a layout and set a `<title>` and `<h1>` in layout as follow:
```vue
<!-- pages/page1.vue -->
<script lang="ts" setup>
definePageMeta({
  title: 'Page 1',
})
</script>

<template>
  <nuxt-layout name="my-layout">
    ...
  </nuxt-layout>
</template>
```

```vue
<!-- layouts/my-layout.vue -->
<script lang="ts" setup>
useHead({ title: useRoute().meta.title })
</script>

<template>
  <div>
    <h1>{{ $route.meta.title }}</h1>
    <slot></slot>
  </div>

</template>
```

And it works great. However, I ran into a problem. On one of the pages the metadata is not static but depends on the result of the fetch
```vue
<!-- pages/profile.vue -->
<script lang="ts" setup>
definePageMeta({
  title: '', // Should be user.name
})

const {data: user} = await useFetch('/api/user')
</script>

<template>
  <nuxt-layout name="my-layout">
    {{ user }}
  </nuxt-layout>
</template>
```
The documentation says:

> `definePageMeta` is a compiler macro. It will be compiled away so you cannot reference it within your component. Instead, the metadata passed to it will be hoisted out of the component. Therefore, the page meta object cannot reference the component (or values defined on the component)

This seems like a common case. But I do not understand how to do it.

**Comment by TheAlexLichter** [comment]
Hey 👋 

as far as I know, you can't use `definePageMeta` in this case. Instead, you might want to save the user in your store, or via `useState`. Then, access the user in your layout and set the title via `useHead`.

**Comment by cawa-93** [comment, marked as answer]
Well, I managed to find, though not very elegant, a solution to the problem. It works due to slots:
```vue
<!-- layouts/my-layout.vue -->
<script lang="ts" setup>
useHead({ title: useRoute().meta.title })
</script>

<template>
  <div>
    <h1><slot name="title">{{ $route.meta.title }}</slot></h1>
    <slot></slot>
  </div>

</template>
```
```vue
<!-- pages/profile.vue -->
<script lang="ts" setup>
const {data: user} = await useFetch('/api/user')
</script>

<template>
  <nuxt-layout name="my-layout">
    <template #title>
      {{ user?.name || '' }}
     </template>
    {{ user }}
  </nuxt-layout>
</template>
```
